File: src/Employees.API/Data/Models/Employee.cs
using Shared.Data;

namespace Employees.API.Data.Models
{
    public class Employee : Entity<long>
    {
        public long Id { get; set; }
        public string StaffId { get; set; }
        public int CompanyId { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;
        public ICollection<Qualification> Qualifications { get; set; } = new List<Qualification>();
        public Company Company { get; set; }
    }
}

File: src/HRMS.Angular/src/app/models/employee.ts
/**
 *             string FirstName,
            string LastName,
            string Email,
            string Address,
            string Phone,
            string Country,

            Qualification
                        string Title,
            string Description,
            string QualificationType,
            string EducationLevel,
            DateTime? DateReceived,
            DateTime? ExpiryDate
*/

export interface EmployeeBase{
    firstName: string,
    lastName: string,
    email: string,
    address: string,
    phone: string,
    country: string,
    position: string,
    department: string
}

export interface Employee extends EmployeeBase{
    id: number,
    staffId: string
}

export interface CreateEmployee extends EmployeeBase{
}

export interface EditEmployee extends EmployeeBase{
    employeeId: number
}

export interface EmployeeDetail extends Employee{
    qualifications: Qualification[]
}

export interface AddQualification{
    employeeId: number,
    title: string,
    description: string,
    qualificationType: number, //Certification, Education
    educationLevel: number,
    dateReceived: Date|null,
    expiryDate: Date|null,
    mediaFile: any
}

export interface Qualification{
    id: number,
    title: string,
    description: string,
    qualificationType: string, //Certification, Education
    educationLevel: string,
    dateReceived: Date|null,
    expiryDate: Date|null,
    mediaFileId: string|null,
    mediaFile: string|null
}